### PR TITLE
Migrate off asylo::StatusOr 

### PR DESCRIPTION
**Pull request by daviddrysdale**
# Checklist

- [x] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

**Comment by daviddrysdale** [comment]
There's still some `Status` fiddling (`grpc::Status` vs `absl::Status` vs `asylo::Status`) which will need another pass later.

**Comment by tiziano88** [comment]
I'm not entirely sure of how much this buys us, I mean we still have those types vendored in third party. Wouldn't it be simpler to keep the asylo dep around until we get rid of the C++ code entirely, and just use `status` from it? AFAICT that compiles without special toolchain support or anything like that.

**Comment by tiziano88** [comment]
Anyways, I'm fine with the change either way, but perhaps @conradgrobler should take a look as he's been more involved in some of this stuff, so I'll leave him to approve.

**Comment by daviddrysdale** [comment]
I think we need to vendor and modify the `asylo::StatusOr` stuff anyway in order to allow logging to shift over to gLog without clashes of preprocessor symbols; it will also be nice to be able to drop the whole Asylo `WORKSPACE` dep.

File: third_party/asylo/status_macros.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+///
+/// Example:
+/// ```
+///   ::absl::Status MultiStepFunction() {
+///     OAK_RETURN_IF_ERROR(Function(args...));
+///     OAK_RETURN_IF_ERROR(foo.Method(args...));
+///     return ::absl::OkStatus();
+///   }
+/// ```
+#define OAK_RETURN_IF_ERROR(expr)                          \
```
Comment (conradgrobler) [inline third_party/asylo/status_macros.h]: Will this rename as a separate commit in the same PR be an issue when the PR is squashed and merged? Or is the plan to merge without squashing?

Comment (daviddrysdale) [reply third_party/asylo/status_macros.h]: I'll rebase-and-merge.

**Comment by daviddrysdale** [comment]
(For #709)

**Pull request merged**